src/daft-io/src/s3_like.rs

```diff
@@ -701,22 +700,25 @@ impl ObjectSource for S3LikeSource {
         }?;
         let key = parsed.path();
 
-        let key = key
-            .strip_prefix('/')
-            .map(|k| k.strip_suffix('/').unwrap_or(k));
-        let key = key.unwrap_or("");
-
+        let key = key.strip_prefix('/').unwrap_or("");
+        let key = if key.is_empty() {
+            "".to_string()
+        } else {
+            let key = key.strip_suffix('/').unwrap_or(key);
+            format!("{key}/")
+        };
```

Comment:
Seems like this might be more simply expressed as:
```rust
let key = key.strip_prefix('/').unwrap_or(key);
let key = if !key.ends_with('/') { format!("{key}/") } else { key };
```

Reply:
What I wanted to take care of here is to drop trailing slashes but ensure that there's atleast one. looks like the right call was actually `trim_**` in https://github.com/Eventual-Inc/Daft/pull/1404/files